core/src/banking_stage/forwarder.rs

```diff
@@ -231,6 +217,28 @@ impl Forwarder {
             )
         });
     }
+
+    fn forward(
+        &self,
+        forward_option: &ForwardOption,
+        packet_vec: Vec<Vec<u8>>,
+        addr: &SocketAddr,
+    ) -> Result<(), TransportError> {
+        match forward_option {
+            ForwardOption::ForwardTpuVote => {
+                // The vote must be forwarded using only UDP.
+                let pkts: Vec<_> = packet_vec.into_iter().zip(repeat(*addr)).collect();
+                batch_send(&self.socket, &pkts).map_err(|err| err.into())
+            }
+            ForwardOption::ForwardTransaction => {
+                // All other transactions can be forwarded using QUIC, get_connection() will use
+                // system wide setting to pick the correct connection object.
+                let conn = self.connection_cache.get_connection(addr);
+                conn.send_data_batch_async(packet_vec)
+            }
+            ForwardOption::NotForward => panic!("should not forward"),
```

Comment:
How confident are we this won't be marked ForwardOption::NotForward? I do not see the caller handle_forwarding checking it. Can we jus log and ignore it?

Reply:
It's checked here: https://github.com/solana-labs/solana/pull/30923/files#diff-028b33dbe7bf459d72bb9d3c1ba754a916e14ad6027728b582754aaecd74b5ccL151

I think a panic is better, calling this without actually forwarding is a logical mistake that should be caught by tests.

Reply:
that said, current tests don't seem sufficient...probably need to add some more that test the different forwarding option(s), instead of just Transactions